sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -202,6 +283,16 @@ private[parquet] class ParquetFilters(pushDownDate: Boolean, pushDownStartWith:
     case ParquetDateType if pushDownDate =>
       (n: String, v: Any) =>
         FilterApi.gtEq(intColumn(n), dateToDays(v.asInstanceOf[Date]).asInstanceOf[Integer])
+
+    case ParquetSchemaType(DECIMAL, INT32, 0, _) if pushDownDecimal =>
```

Comment:
Why match 0 instead of _?

Reply:
In fact, the length is always 0, I replaced it to `_`.